packages/react-compose/src/types.ts

```diff
@@ -84,9 +84,9 @@ export type GenericDictionary = Record<string, any>;
  */
 export type ClassDictionary = Record<string, string>;
 
-export type ComposePreparedOptions<TProps = {}> = {
+export type ComposePreparedOptions<TProps = {}, TState = TProps> = {
   className: string;
-  classes: ClassDictionary;
+  classes: (undefined | ClassDictionary | ((state: GenericDictionary, slots: GenericDictionary) => ClassDictionary))[];
```

Comment:
define type `ClassFunction = (state: GenericDictionary, slots: GenericDictionary) => ClassDictionary)` and use both here and `createClassResolver`

Reply:
done

Reply:
done